rdtools/availability.py

```diff
@@ -0,0 +1,620 @@
+       inverter downtime analysis", 2020 IEEE 47th Photovoltaic Specialists
+       Conference (PVSC).
+    """
+
+    def __init__(self, power_system, power_subsystem, energy_cumulative,
+                 power_expected):
+        for series in [power_subsystem, energy_cumulative, power_expected]:
+            if not power_system.index.equals(series.index):
+                raise ValueError("Input timeseries indexes must match")
+
+        self.power_system = power_system
+        self.power_subsystem = power_subsystem
+        self.energy_cumulative = energy_cumulative
+        self.power_expected = power_expected
+
+    def _calc_loss_subsystem(self, low_threshold, relative_sizes,
+                             power_system_limit):
+        """
+        Estimate timeseries production loss from subsystem downtime events.
+
+        This implements the "power comparison" method from [1]_ of comparing
+        subsystem power data to total system power (e.g. inverter power to
+        meter power).
+
+        Because this method is based on peer-to-peer comparison at each
+        timestamp, it is not suitable for full system outages (i.e., at least
+        one inverter must be reporting along with the system meter).
+
+        Sets the `reporting_mask` and `loss_subsystem` attributes.
+
+        Parameters
+        ----------
+        low_threshold : float or pd.Series
+            An optional threshold used to naively classify subsystems as
+            online. If the threshold is a scalar, it will be used for all
+            subsystems. For subsystems with different capacities, a pandas
+            Series may be passed with index values matching the columns in
+            ``power_subsystem``. Units must match ``power_subsystem`` and
+            ``power_system``. If omitted, the limit is calculated for each
+            subsystem independently as 0.001 times the 99th percentile of its
+            power data.
+
+        relative_sizes : dict or pd.Series
+            The production capacity of each subsystem, normalized by the mean
+            subsystem capacity. If not specified, it will be estimated from
+            power data.
+
+        power_system_limit : float or pd.Series, optional
+            Maximum allowable system power. This parameter is used to account
+            for cases where online subsystems can partially mitigate the loss
+            of an offline subsystem, for example a system with a plant
+            controller and dynamic inverter setpoints. This constraint is
+            only applied to the subsystem loss calculation.
+        """
+        power_subsystem = self.power_subsystem
+        power_system = self.power_system
+        power_subsystem = power_subsystem.fillna(0)
+        power_system = power_system.clip(lower=0)
+
+        # Part A
+        if low_threshold is None:
+            low_threshold = power_subsystem.quantile(0.99) / 1000
```

Comment:
`/ 1000` I initially assumed is a unit conversion but after `looks_online` I take it that `/ 1000` is a way to set a lower threshold, presumably `.quantile(0.01)` or similar is unreliable (could be 0 if there are outages)? Maybe `power_subsystem[power_subsystem.nonzero].quantile(0.01)` just a thought.

Maybe worth a comment.

Reply:
> I take it that / 1000 is a way to set a lower threshold

Correct -- not a unit conversion, just 0.1% of the 99th percentile.  It's done that way to defend against various low-power issues:  outages like you suggest, nighttime values, dawn/dusk shading, snow coverage... If there was a way to reliably filter out abnormal behavior then I'd be fine with `filtered_subset.quantile(0.01)` like you suggest, but that filter seems difficult to construct.  For example some inverters report small nonzero numbers (positive or negative) at night, so just checking `.nonzero` wouldn't work for them.   In contrast, it's much more likely for the high-power data to reflect normal behavior, so that seemed like the most reliable way to calculate this value.  The 99th percentile is just to avoid the rare non-physical data spike you sometimes see in field datasets. 

I agree it warrants a comment.  Thanks!